src/app/commissioner_app.cpp

```diff
@@ -1337,18 +1337,19 @@ void CommissionerApp::HandleEnergyReport(const std::string *aPeerAddr,
                                          const ByteArray *  aEnergyList,
                                          Error              aError)
 {
-    if (aError != Error::kNone)
-    {
-        return;
-    }
-
+    Error   error;
     Address addr;
-    addr.Set(*aPeerAddr);
-    ASSERT(addr.IsValid());
+
+    SuccessOrExit(error = aError);
```

Comment:
```suggestion
    SuccessOrExit(aError);
```
Is it ok?

Reply:
I agree. I don't we need the `error` variable here on next line since we are not using it after this.
```
SuccessOrExit(addr.Set(*aPeerAddr));
```